app/Http/Controllers/RegistrationController.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Lesson;
+use App\Models\PaymentStructure;
+use App\Models\Registration;
+use App\Models\User;
+use Illuminate\Contracts\Support\Renderable;
+use Illuminate\Foundation\Application;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Routing\Redirector;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Date;
+
+class RegistrationController extends Controller
+{
+
+    /**
+     * Show the view for signing up for a lesson
+     *
+     * @param int $id
+     * @return Application|RedirectResponse|Redirector|Renderable
+     */
+    public function userIndex(): Application|RedirectResponse|Redirector|Renderable
+    {
+        $user = Auth::user();
+        if (!$user){
+            return redirect(route('lesson.index'))->withErrors(__('registration.public_signup_errors_hasToBeLoggedIn'));
```

Comment:
Because then it is clear for the user how to sign up, but when they then try they are prompted to sig up
![image](https://github.com/Darchie4/AMT-on-Laravel/assets/50356144/fe80edfb-5d17-444f-b1ab-729a884ebd38)

If the button was hidden for the user it would be unclear how to sign up for lessons / that they need to login before they can sign up